### PR TITLE
[TOOLS-167] Removing cache from followup pivot tables

### PR DESCRIPTION
**Pull request by JonathanNasc**

### CODE REVIEW

File: src/main/java/objective/taskboard/followup/impl/FollowUpDataProviderFromCurrentState.java [hunk 1]
```diff
@@ -313,17 +302,17 @@ private FollowUpData createSubTaskFollowup(Issue demand, Issue task, Issue subta
         followUpData.taskDescription = issueDescription(task);
         followUpData.taskFullDescription = issueFullDescription(task);
         followUpData.taskRelease = coalesce(getRelease(subtask), getRelease(task), getRelease(demand), "No release set");
-        followUpData.taskOrder = task.getStatusPriority() + taskPriorityOrder;
+        followUpData.taskOrder = task.getOrder();
         
         followUpData.subtaskType = subtask.getIssueTypeName();
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/followup/impl/FollowUpDataProviderFromCurrentState.java]: MAJOR A "NullPointerException" could be thrown; "subtask" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/main/java/objective/taskboard/data/Issue.java [hunk 2]
```diff
@@ -377,9 +377,15 @@ public String getStatusName() {
     public String getStatusOrderedName() {
         return getStatusPriority() + "." + getStatusName();
     }
+    
+    @JsonIgnore
+    public String getOrder() {
+        String priorityOrder = getPriorityOrder() != null ? getPriorityOrder().toString() : "";
+        return getStatusPriority() + priorityOrder;
```
Comment (taksan) [inline src/main/java/objective/taskboard/data/Issue.java]: Esta forma de determinar a prioridade está incorreta. Exemplo:
Issue A
StatusPriority 1, priorityOrder 1000 -> order "1:1000"
Issue B
StatusPriority 1
priorityOrder 2
StatusPriority 1, priorityOrder 2 -> order "1:2"

Nesse caso a Issue "B" deveria estar antes da "A", porém a comparação de string vai considerar o contrário

File: src/main/java/objective/taskboard/followup/FollowUpData.java [hunk 3]
```diff
@@ -61,6 +61,10 @@
 
     public String queryType;
     
+    public String getOrder() {
```
Comment (taksan) [inline src/main/java/objective/taskboard/followup/FollowUpData.java]: Comparação de ordem por string vai trazer problemas similares ao indicado anteriormente.

File: src/main/java/objective/taskboard/followup/impl/FollowUpDataProviderFromCurrentState.java [hunk 4]
```diff
@@ -313,17 +312,19 @@ private FollowUpData createSubTaskFollowup(Issue demand, Issue task, Issue subta
         followUpData.taskDescription = issueDescription(task);
         followUpData.taskFullDescription = issueFullDescription(task);
         followUpData.taskRelease = coalesce(getRelease(subtask), getRelease(task), getRelease(demand), "No release set");
-        followUpData.taskOrder = task.getStatusPriority() + taskPriorityOrder;
+        followUpData.taskStatusPriority = task.getStatusPriority();
+        followUpData.taskPriorityOrder = task.getPriorityOrder();
         
         followUpData.subtaskType = subtask.getIssueTypeName();
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/followup/impl/FollowUpDataProviderFromCurrentState.java]: MAJOR A "NullPointerException" could be thrown; "subtask" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

**Pull request merged**